### PR TITLE
[Woo POS] Voiceover improvements

### PR DESCRIPTION
**Pull request by joshheald**
Part of: #13517

## Description

This PR improves the Voiceover support of POS.

This involves quite a few small tweaks – you may find it helpful to go commit-by-commit.

### Still to do:
 - POS Modal accessibility support – it's really difficult to navigate modal content with VoiceOver, but I'll deal with this on a separate PR

## Steps to reproduce

1. Launch the app and enter POS mode – note, do this with VoiceOver off, as there's a bug accessing items using VoiceOver on the hub menu.
2. Turn VoiceOver on (set your device's accessibility shortcut to do this with 3 presses on the side button)
3. Navigate the app and check that it is accessible – 
 
### VoiceOver refresher
If you can't remember how to use it, it's worth reading up again, but here's some basics:

Swipe up and down to use heading navigation.
Swipe left or right to go back or forward one item.
Double tap to activate buttons.

## Testing information

Limited unit tests as almost all the changes are to the views.

Tested on iPad only because this is POS.

Things I tested:
- [x] Navigation using headers.
- [x] Images are either hidden from VoiceOver, or described accurately.
- [x] Rows, totals, and messages are grouped together or combined into a single element where that makes sense.
- [x] Items in lists are announced under their column – these are grouped together.
- [x] Card payment views are announced in a group.
- [x] Floating controls are accessible with VoiceOver

I merged trunk, and had some conflicts to fix. Most were straightforward, but I tweaked the behaviour of the Cart header in larger accessibility sizes, to show as follows:

```
< Cart     2 items
               | Clear |
````

It's worth a quick check with larger fonts as you test, because there was some manual merging I had to do.

## Screenshots

https://github.com/user-attachments/assets/107c33b7-a8a1-49cb-83e4-50ec2a2e4a8a

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/Reader Messages/PointOfSaleCardPresentPaymentDisplayReaderMessageMessageView.swift [hunk 1]
```diff
@@ -19,6 +19,7 @@ struct PointOfSaleCardPresentPaymentDisplayReaderMessageMessageView: View {
                 Text(viewModel.message)
                     .font(.posTitleEmphasized)
                     .foregroundStyle(.white)
+                    .accessibilityAddTraits(.isHeader)
```
Comment (staskus) [inline WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/Reader Messages/PointOfSaleCardPresentPaymentDisplayReaderMessageMessageView.swift:22]: Why do we set the message trait to the header? And use it so widely in other places?

Docs say:

> The accessibility element is a header that divides content into sections, like the title of a navigation bar.

https://developer.apple.com/documentation/swiftui/accessibilitytraits/isheader

I would say mostly all of it is `.staticText`?

File: WooCommerce/Classes/POS/Presentation/CartView.swift [hunk 2]
```diff
@@ -19,46 +19,54 @@ struct CartView: View {
...
                         .shimmering(active: viewModel.isAddMoreDisabled)
-                    Text(Localization.cartTitle)
-                        .font(Constants.primaryFont)
-                        .foregroundColor(cartViewModel.cartLabelColor)
-                }
 
-                Spacer()
-                    .renderedIf(!dynamicTypeSize.isAccessibilitySize)
-                if let itemsInCartLabel = cartViewModel.itemsInCartLabel {
                     HStack {
```
Comment (staskus) [inline WooCommerce/Classes/POS/Presentation/CartView.swift:23]: These changes are okay 👍 We just structure cart header with item count together now.

**Review by staskus — APPROVED** [review]
✅ Looks good. I tested on both device and simulator with an inspector. I think any further improvements could be made in future tasks.

I have one bigger question about using "header" trait everywhere.

And I think it would make sense to consider focusing on accessibility elements when the state changes, especially when payment status or errors appear.

File: WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/Reader Messages/PointOfSaleCardPresentPaymentDisplayReaderMessageMessageView.swift (hunk 1 above)
Comment (joshheald) [reply WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/Reader Messages/PointOfSaleCardPresentPaymentDisplayReaderMessageMessageView.swift:22]: Headers are important for navigation – for most users of the app, that's the visual "navigation" of the screen, seeing what's important. For a VoiceOver user, it's swiping around the screen and quickly identifying the important sections.

I generally follow this approach:

> Header
> Any visually distinct header text should have this trait applied. 
– https://mobilea11y.com/blog/traits/#header

It is a balance, as adding the "Heading" announcement to a lot of elements can get noisy, but I don't think we have many heading elements on screen at the same time (3 maximum, I think.)

The most questionable one I've done is adding it to the `Check out` button... but it's so useful to be able to do a single vertical swipe between Products, Cart, and Check out, that I think it's the right approach.

**Comment by joshheald** [comment]
> And I think it would make sense to consider focusing on accessibility elements when the state changes, especially when payment status or errors appear.

Agreed. I took a look to see whether I could do it quickly in the context of this ticket, but it wasn't straightforward. I'll make a ticket, but I think we should consider it post-POC.

File: WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/Reader Messages/PointOfSaleCardPresentPaymentDisplayReaderMessageMessageView.swift (hunk 1 above)
Comment (staskus) [reply WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/Reader Messages/PointOfSaleCardPresentPaymentDisplayReaderMessageMessageView.swift:22]: Got it, makes sense. I think we'll be only able to perfect all of this after we get specific feedback.

**Pull request merged**